### ISSUE TITLE
Browsable Time Slider

### ISSUE THREAD

**Issue by IshaGupta18** [publiclab/simple-data-grapher#22]
I was trying to make the browsable time slider using https://github.com/schme16/Chart.js-RangeSlider .

But when I included the script file `https://gitcdn.link/cdn/schme16/Chart.js-RangeSlider/677a7eb6c295772606d300697eaf552245e7f171/dist/RangeSlider-all.min.js`, it seems to be interfering with the bootstrap's  JS file and shows an error like 

![image](https://user-images.githubusercontent.com/40794215/57913439-40e47c00-78aa-11e9-8aa2-101dc1d26d1b.png)

What could be the reason that the scripts are interfering with each other? Is there a way to fix it, because I tried but couldn't, we can keep trying or any other suggestions for time sliders? @jywarren @namangupta01 Thank You!

**Comment by IshaGupta18** [comment]
Okay, so I updated the code and found out that I was accidentally plotting the sample data instead of the complete data. Apologies for that. The graphs will now look more cluttered like this:
![image](https://user-images.githubusercontent.com/40794215/57964943-b0b23f80-795a-11e9-9ab8-b3f2b6f76edf.png)


So we will need to think some stuff through. The requirement for a range slider becomes an absolute necessity now and we will have to refine the graph UI for best results. Kindly give your views on it. Thank you!

**Comment by namangupta01** [comment]
I think if that is occurring with the bootstrap, we can't fix it. In that case, it would be better to use something else than bootstrap?

**Comment by namangupta01** [comment]
But as I see the error is not due to Bootstrap. This would work super.

**Comment by IshaGupta18** [comment]
But that would break the consistency of the sites we have at PL, all use
bootstrap, can we think of an alternative for the range slider?

**Comment by IshaGupta18** [comment]
So what is causing this error? Because it's causing a conflict.

**Comment by namangupta01** [comment]
We are making this library standalone which means it would be used by the public lab as well as any other users also. So it would be better if we give them the freedom whether they want to use or not. We should give a default design and if the user of this library wants to improve the css he can do so.

**Comment by namangupta01** [comment]
Don't think of only using this for PublicLab anyone should be able to use it on their website. Right?
What do you think?

**Comment by IshaGupta18** [comment]
Yes agreed! That would be great, although we still will be needing
bootstrap here for some of the designing, it's a good asset to give up.
Let's figure out why this error is being caused

**Comment by IshaGupta18** [comment]
@jywarren @namangupta01  what do you think might be causing the problem here?

**Comment by IshaGupta18** [comment]
@gauravano @SidharthBansal @souravirus kindly help me out here a little!

**Comment by gauravano** [comment]
@IshaGupta18 are you still facing this issue?

**Comment by gauravano** [comment]
Also, can you please push the sample CSV file so that everyone can be on the same page format wise? Thanks!

**Comment by gauravano** [comment]
I have to press upload every time to plot the graph so how about having an explicit button for that? Also, I wasn't able to test upload using the URL as I am getting CORS. I tried using - http://insight.dev.schoolwires.com/HelpAssets/C2Assets/C2Files/C2ImportCalEventSample.csv.

**Comment by IshaGupta18** [comment]
Yes sure! Let me add the sample csv here. And yes this issue persists.

**Comment by IshaGupta18** [comment]
hey, this file works for me! I think it didn't work for you because the latest changes and few bugs were fixed in the PRs I have raised.
![image](https://user-images.githubusercontent.com/40794215/58105518-43c1d280-7c04-11e9-85cc-5688167e74db.png)

![image](https://user-images.githubusercontent.com/40794215/58105636-779cf800-7c04-11e9-90e2-2a6091f97d3e.png)

**Comment by IshaGupta18** [comment]
@gauravano what do you think?

**Comment by IshaGupta18** [comment]
Okay so I have fixed this carousel bug! yay! But here's the new one which is creating a problem

![image](https://user-images.githubusercontent.com/40794215/58492630-e93dee80-818e-11e9-8673-73dfabb66a6a.png)

Here's the snippet of the code causing it:

![image](https://user-images.githubusercontent.com/40794215/58492712-17233300-818f-11e9-851e-de77d88fb315.png)

So basically the range-slider is not getting initialized properly.

Although, I have written it very similar to the working fiddle here: http://jsfiddle.net/schme16/xfyvvup8/

What do you think is creating a problem? I'll push the code if you need it! I will really appreciate some help here! Thank you

**Comment by Souravirus** [comment]
Hi @IshaGupta18, can you send larger screenshots as these are not visible. I think that if you push the code somewhere or send it in a gist, that would be really great.
Thank you!!

**Comment by IshaGupta18** [comment]
Yes absolutely!

**Comment by IshaGupta18** [comment]
#31 will help you with the code @Souravirus. Thanks a lot!
